### PR TITLE
Bookstore: Redux Store (Use Redux in Components)

### PR DESCRIPTION
**Pull request by ZikaZaki**
# Bookstore: Redux Store (Use Redux in Components)

## In this pull request I did the followings:

- [x] Understood the concept of state management.
- [x] Used store, actions and reducers in React.
- [x] Connect React and Redux using `Provider`. 
- [x] Made sure that the state is immutable in reducers!
- [x] Changed the default state in the books reducer from an empty array to an array with a few books.
- [x] Imported Redux `Provider` and the store in the main component of the React app.
- [x] Connected the app to the Redux store.
- [x] Made sure that list of books displays books from Redux store.
   - Used `useSelector` to consume the state.
- [x] In the React component responsible for adding new books, I added functionality for "Add new" button click:
   - Imported Redux `useDispatch`.
   - Set the data inputs in the local React state (set title and author).
   - Took the inputs from the state and generated a unique id and added them to an object.
   - Once the new book object is ready to be submitted to Redux store, dispatched a corresponding action.
- [x] In the React component responsible for removing books, add a functionality for "Remove" button click:
   - Imported Redux `useDispatch`.
   - Dispatched a corresponding action.
- [x] In the React component responsible for categories, add a functionality for "Check status" button click:
   - Imported Redux `useDispatch`.
   - Dispatched a corresponding action.
- [x] Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Made sure that the project is perfectly working as intended.
- [x] Documented the project professionally.

----
## [❗ Optional ] 
- [x] Added some styles to improve the look and make it easier for the code review to check the application.

### CODE REVIEW

File: src/components/BookList/BookList.js [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
-  const [books] = useState(getInitialBooks());
+  // function getInitialBooks() {
+  //   const books = booksReducer(undefined, { type: 'books/GET_BOOKS' });
+  //   if (books) {
+  //     return books;
+  //   }
+  //   return [];
+  // }
+
+  // const [books] = useState(getInitialBooks());
```
Comment (victorokoroji) [inline src/components/BookList/BookList.js]: - [ ] Kindly remove all commented code from your codebase to make your codebase clean and readable 👍🏽

File: src/components/BookList/BookList.js [hunk 2]
```diff
@@ -1,30 +1,29 @@
-import React, { useState } from 'react';
-import { v4 as uuidv4 } from 'uuid';
+import React from 'react';
+import { useSelector } from 'react-redux';
+// import booksReducer from '../../redux/books/books';
```
Comment (victorokoroji) [inline src/components/BookList/BookList.js]: - [ ] Kindly remove all commented code from your codebase to make your codebase clean and readable 👍🏽

File: src/index.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
 import ReactDOM from 'react-dom/client';
+import store from './redux/configureStore';
 import './index.css';
 import App from './App';
 // import BooksContainer from './components/BooksContainer';
```
Comment (victorokoroji) [inline src/index.js]: - [ ] Kindly remove all commented code from your codebase to make your codebase clean and readable 👍🏽

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @ZikaZaki, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (ZikaZaki) [reply src/index.js]: Thanks @vickymarz  for your professionalism!

**Review by DammyShittu — APPROVED** [review]
Hi @ZikaZaki,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**